server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -437,6 +438,52 @@ public void testReplicationAfterForceMerge() throws Exception {
         }
     }
 
+    public void testReplicationAfterForceMergeOnPrimaryShardsOnly() throws Exception {
```

Comment:
@gaobinlong You can refactor this and `testReplicationAfterForceMerge` into a common method something like below and then call it from both the test cases:

```
public void testReplicationAfterForceMerge() throws Exception {
      performReplicationAfterForceMerge(false, SHARD_COUNT*REPLICA_COUNT);
}

public void testReplicationAfterForceMergeOnPrimaryShardsOnly() throws Exception {
      performReplicationAfterForceMerge(true, SHARD_COUNT);
}


private void performReplicationAfterForceMerge(boolean primaryOnly, int expectedSuccessfulShards) {
        final String nodeA = internalCluster().startDataOnlyNode();
        final String nodeB = internalCluster().startDataOnlyNode();
        createIndex(INDEX_NAME);
        ensureGreen(INDEX_NAME);

        final int initialDocCount = scaledRandomIntBetween(0, 10);
        final int additionalDocCount = scaledRandomIntBetween(0, 10);
        final int expectedHitCount = initialDocCount + additionalDocCount;
        try (
            BackgroundIndexer indexer = new BackgroundIndexer(
                INDEX_NAME,
                "_doc",
                client(),
                -1,
                RandomizedTest.scaledRandomIntBetween(2, 5),
                false,
                random()
            )
        ) {
            indexer.start(initialDocCount);
            waitForDocs(initialDocCount, indexer);

            flush(INDEX_NAME);
            waitForSearchableDocs(initialDocCount, nodeA, nodeB);

            // Index a second set of docs so we can merge into one segment.
            indexer.start(additionalDocCount);
            waitForDocs(expectedHitCount, indexer);
            waitForSearchableDocs(expectedHitCount, nodeA, nodeB);

            // Perform force merge
            final ForceMergeResponse forceMergeResponse = client().admin()
                .indices()
                .prepareForceMerge(INDEX_NAME)
                .setPrimaryOnly(primaryOnly)
                .setMaxNumSegments(1)
                .setFlush(false)
                .get();
            assertThat(forceMergeResponse.getFailedShards(), is(0));
            assertThat(forceMergeResponse.getSuccessfulShards(), is(expectedSuccessfulShards));
            refresh(INDEX_NAME);
            verifyStoreContent();
        }
}
```

Reply:
Thanks, I've done that.